asset/boltdb_manager.go

```diff
@@ -133,7 +134,16 @@ func (b *boltDBAssetManager) Get(ctx context.Context, asset *corev2.Asset) (*Run
 
 		// verify
 		if err := b.verifier.Verify(tmpFile, asset.Sha512); err != nil {
-			return err
+			// Attempt to retrieve the size of the downloaded asset
+			var size int64
+			if fileInfo, err := tmpFile.Stat(); err == nil {
+				size = fileInfo.Size() / 1000 // kilobytes
+			}
+
+			return fmt.Errorf(
+				"could not validate downloaded asset %q (%d kilobytes): %s",
```

Comment:
Maybe a good use case for https://godoc.org/github.com/dustin/go-humanize#Bytes ?

Reply:
Updated with this lib, much better 👌 
```
Output:    error getting assets for check: could not validate downloaded asset "sensu-go-hello-world" (5.8 MB): sha512 of downloaded asset (8b43b938264a96b236fbec55814d34745cd695b092a9806e21f58f041d8a00a56e8aeadb1bdeb26e0612b332b51c998d3273906e045add4c14c5173f49b2aa3b) does not match specified sha512 in asset definition (06665fea5b7c7ee15e7822820aa7ddb791cc6338e38444e976e601bc7d7970592e806a7b88733690a238b7325437d31f85e98ae2fe47b008ca09c86530da9600)
```